rclpy/src/rclpy_common/src/common.c

```diff
@@ -273,3 +273,90 @@ rclpy_convert_to_py(void * message, PyObject * pyclass)
   }
   return convert(message);
 }
+
+PyObject * _create_event_payload_container(const char * class_name, PyObject * args)
+{
+  PyObject * pyqos_event_module = NULL;
+  PyObject * pyqos_event_class = NULL;
+  PyObject * pyqos_event = NULL;
+
+  pyqos_event_module = PyImport_ImportModule("rclpy.qos_event");
```

Comment:
Let's have a single cleanup logic using Py_XDECREF on all variables to reduce the complexity of this code.

Reply:
ok - I'll restructure it a bit